catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -147,13 +157,44 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
       query.setFacetMinCount(textFacetProp.getMinFacetCount());
     }
 
-    long totalHits;
-    List<Result> results = new ArrayList<>();
-    Map<String, Serializable> responseProps = new HashMap<>();
+    Serializable suggestQueryProp = request.getPropertyValue(EXPERIMENTAL_SUGGESTION_REQUEST_KEY);
```

Comment:
We need to go straight to Solr APIs to get solr specific suggestion capabilities?  Is this something that we need to think about going through DDF for in 3.0 ?

Reply:
Yeah, the suggester is configured as a separate query handler. It builds its own index as well. It depends on what kind of suggestion functionality we want to support in 3.0, as it's likely not simple or even feasible to have an arbitrary "build a suggester for this field" functionality, as the solr config needs to contain the suggester component configuration. I'm not sure what our goals are with suggestions.